src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,97 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int VALUE_INDEX = 1;
+    private static final int WORD_INDEX = 0;
+    private static final int SUPPLY_INDEX = 0;
+    private static final int BUY_INDEX = 1;
+    private static final int RESULT_INDEX = 2;
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT_VALUE = "result";
+    private static final String ZERO_VAR = "0";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String statistic = readText(fromFileName);
+        writeText(toFileName, statistic);
+    }
+
+    private void writeText(String toFileName, String statistic) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            bufferedWriter.write(statistic);
+        } catch (IOException e) {
+            throw new RuntimeException("can't write to file");
+        }
+    }
+
+    private String readText(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            return getResultText(bufferedReader);
+        } catch (IOException e) {
+            throw new RuntimeException("wrong file path");
+        }
+    }
+
+    private String getResultText(BufferedReader bufferedReader) throws IOException {
+        StringBuilder list = new StringBuilder();
+        String[][] information = getMapInformation(bufferedReader);
+        for (int i = 0; i < information.length; i++) {
+            if (i != 0) {
+                list.append(System.lineSeparator());
+            }
+            list.append(information[i][WORD_INDEX]).append(",")
+                    .append(information[i][VALUE_INDEX]);
+        }
+        return list.toString();
+    }
+
+    private String[][] getMapInformation(BufferedReader bufferedReader)
```

Comment:
we should not have methods that are receiving the reader as a parameter, and as it is written in the common mistakes we should close the reader/writer resources as soon as possible, so i n general I would break down our logic into 3 methods:
* String readFile(String fromFileName)
* String createReport(String data)
* void writeToFile (String report)